### DISCUSSION TITLE
Correction to trivy.repo creation code snippet | #2099 bugfix

### DISCUSSION THREAD

**Discussion by 02fa** [aquasecurity/trivy#4545] [Documentation]
### Description

With reference to [#2099](https://github.com/aquasecurity/trivy/issues/2099), the solution doesn't quite work.

1=
https://github.com/aquasecurity/trivy/blob/854b63940a530155b70c94787c34d922ee2ef4b4/docs/getting-started/installation.md?plain=1#L13

By fixing release version to the current target host readings we may hit an issue, when the host OS is dist-upgraded.

2=
https://github.com/aquasecurity/trivy/blob/854b63940a530155b70c94787c34d922ee2ef4b4/docs/getting-started/installation.md?plain=1#L17

heredoc is implicitly double quoted. This results in undefined `$basearch` expansion into an empty string. By escaping `\$basearch` on .repo file we are getting error: `Error: Failed to download metadata for repo 'trivy': Cannot download repomd.xml: Cannot download repodata/repomd.xml`. Correct way of escaping `$basearch` is to escape "limit string" `'EOF'` itself. Single and double quotes, along with \ will do (scroll down to [Example 19-7. Parameter substitution turned off](https://tldp.org/LDP/abs/html/here-docs.html)).

Corrected here:
https://github.com/Wang-W-e-i/trivy/blob/0f2b2107ee0848c3f05b4becffdae5d3fe085ae8/docs/getting-started/installation.md?plain=1#L13-L21

3=
https://aquasecurity.github.io/trivy-repo/ may need to be updated in a similar way.

4=
It must be my mirror or something :o(. It doesn't install via RPM for some reason.

Is Trivy RPM repository maintained? Or is it a temporary glitch with my mirror? How much does it cost to extend repository to ARM64 architecture, please?

`sudo dnf install trivy
<snip>
Error: Error downloading packages:
  trivy-0.42.0-1.x86_64: Download failed: Downloading successful, but checksum doesn't match. Calculated: 238cd36e7685e32eda85a6ef003a6064b39948a1c35a3d3fbf51e6b0af5e4dc7(sha256)  Expected: f7c1f97538d02de4db29f51ac596d8cfe5f1a21a517825ef08a14f35d5224363(sha256)`

### Link

https://github.com/aquasecurity/trivy/blob/854b63940a530155b70c94787c34d922ee2ef4b4/docs/getting-started/installation.md?plain=1#L12-L24

### Suggestions

Fix as per this proposal
https://github.com/Wang-W-e-i/trivy/blob/0f2b2107ee0848c3f05b4becffdae5d3fe085ae8/docs/getting-started/installation.md?plain=1#L13-L21

**Comment by knqyf263** [comment]
@afdesk Can you please take a look?

**Comment by afdesk** [comment]
@Wang-W-e-i thanks for the report! 
yes, we have an issue with the latest deploy to RPM repository. 
but I hope we fixed it.
Could you confirm that it works for you?

also, as wrote @knqyf263 we changed an approach after #4077 
so `RELEASE_VERSION` (`$releasever`) doesn't need anymore.
wdyt?

**Reply by knqyf263** [reply]
Oh, yes. Did we forget to update the installation instruction? I thought we removed $releasever.

**Reply by afdesk** [reply]
yeh, forgot...

**Reply by knqyf263** [reply]
OK, let's remove it, then.

**Reply by 02fa** [reply]
Thank you, @afdesk ! Trivy did install on ARM64 via RPM. This addresses points 1= and 4=. 

Proper`\$basearch` escaping (2=) and other doc update (3=) remain open.

**Reply by knqyf263** [reply]
@afdesk Would you update the doc?

**Reply by afdesk** [reply]
@knqyf263 Yes, sure. i'll update the doc today. I remember it.

**Reply by afdesk** [reply]
@Wang-W-e-i I'll try to resolve (2=) and (3=) too

**Reply by afdesk** [reply]
I want to take #4547 as a base for it.

**Reply by afdesk** [reply]
@Wang-W-e-i thanks for your help and comments!

I've created a PR to fix (=3) https://github.com/aquasecurity/trivy-repo/pull/30.

to fix (=2) let's use #4547 

and we can close this discussion, right?

**Reply by 02fa** [reply]
PR looks perfect, thank you!

**Comment by 02fa** [comment]
Moved to https://github.com/aquasecurity/trivy-repo/pull/30 and https://github.com/aquasecurity/trivy/pull/4547
